core/src/main/scala/org/apache/spark/status/LiveEntity.scala

```diff
@@ -524,30 +648,35 @@ private class LiveRDD(val info: RDDInfo) extends LiveEntity {
 
   import LiveEntityHelpers._
 
-  var storageLevel: String = weakIntern(info.storageLevel.description)
+  private val storageLevel = weakIntern(info.storageLevel.description)
   var memoryUsed = 0L
   var diskUsed = 0L
 
-  private val partitions = new HashMap[String, LiveRDDPartition]()
+  private val partitions = new OpenHashMap[Int, LiveRDDPartition]()
   private val partitionSeq = new RDDPartitionSeq()
+  private var partitionCount = 0
```

Comment:
Why a separate `partitionCount` is needed? The `partitionSeq` has a length method which follows the add/remove partition.